mm/ktsan/sync_mtx.c

```diff
@@ -4,14 +4,31 @@
 
 void kt_mtx_pre_lock(kt_thr_t *thr, uptr_t pc, uptr_t addr, bool wr, bool try)
 {
+	if (thr->mtx_depth++ != 0)
+		return;
+
+	kt_thr_event_disable(thr);
```

Comment:
Replace this with:

	if (thr->event_disable_depth++ != 0)
            return;